BE/toDoListAPI/src/main/java/school/mindera/toDoListAPI/service/TaskService.java

```diff
@@ -23,12 +25,88 @@ public TaskService(TasksRepository tasksRepository, UsersRepository usersReposit
         this.usersRepository = usersRepository;
     }
 
+    public ResponseEntity<List<DTOTaskPreview>> getTaskPreview(Integer userId) {
+        Optional<UsersEntity> user = usersRepository.findById(userId);
+
+        if(user.isEmpty()){
+            throw new InvalidUserException("Invalid user");
+        }
+
+        Optional<List<TasksEntity>> tasks = tasksRepository.findByUserId(user.get());
+
+        if(tasks.isEmpty()){
+            return ResponseEntity.ok(new ArrayList<>());
+        }
+
+        List<DTOTaskPreview> dtos = new ArrayList<>();
+
+        tasks.get().forEach(task -> {
+            DTOTaskPreview dto = Converter.toDTOTaskPreview(task);
+            dtos.add(dto);
+        });
```

Comment:
acho que tu pode fazer isso só numa linha
```suggestion
        tasks.get().forEach(task -> {
            dtos.add(Converter.toDTOTaskPreview(task));
        });
```